src/libUtils/EvmUtils.cpp

```diff
@@ -0,0 +1,182 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+#include <array>
+#include <iostream>
+#include <mutex>
+#include <string>
+#include <unordered_map>
+
+#include <boost/filesystem.hpp>
+#include "EvmUtils.h"
+
+#include "JsonUtils.h"
+#include "Logger.h"
+#include "common/Constants.h"
+#include "libData/AccountData/Account.h"
+#include "libData/AccountData/TransactionReceipt.h"
+#include "libPersistence/ContractStorage.h"
+#include "libUtils/EvmCallParameters.h"
+#include "libUtils/EvmJsonResponse.h"
+
+using namespace std;
+using namespace boost::multiprecision;
+
+Json::Value EvmUtils::GetEvmCallJson(const EvmCallParameters& params) {
+  Json::Value arr_ret(Json::arrayValue);
+
+  arr_ret.append(params.m_contract);
+  arr_ret.append(params.m_caller);
+  std::string code;
+  try {
+    // take off the EVM prefix
+    std::copy(params.m_code.begin() + 3, params.m_code.end(),
+              std::back_inserter(code));
+    arr_ret.append(code);
+  } catch (std::exception& e) {
+    arr_ret.append(params.m_code);
+  }
+  arr_ret.append(params.m_data);
+  arr_ret.append(params.m_apparent_value.str());
+  arr_ret.append(Json::Value::UInt64(params.m_available_gas));
+
+  if (LOG_SC) {
+    LOG_GENERAL(WARNING, "Sending to EVM-DS" << arr_ret);
+  }
+
+  return arr_ret;
+}
+
+bool EvmUtils::EvmUpdateContractStateAndAccount(
+    Account* contractAccount, evmproj::ApplyInstructions& op) {
+  if (op.OperationType() == "modify") {
+    // Reset State for this contract
+    try {
+      if (op.isResetStorage()) {
+        //
+        // Reset Meta Data for this Address effectively clears down the contract
+        // storage for this Contract
+        //
+        std::map<std::string, bytes> t_metadata;
+
+        t_metadata.emplace(
+            Contract::ContractStorage::GenerateStorageKey(
+                Address(op.Address()), CONTRACT_ADDR_INDICATOR, {}),
+            Address(op.Address()).asBytes());
+
+        if (!contractAccount->UpdateStates(Address(op.Address()), t_metadata,
+                                           {}, true)) {
+          LOG_GENERAL(WARNING,
+                      "Account::UpdateStates reset metaData and Merkyle tree");
+        }
+        contractAccount->SetStorageRoot(dev::h256());
+      }
+    } catch (std::exception& e) {
+      // for now catch any generic exceptions and report them
+      // will exmine exact possibilities and catch specific exceptions.
+      LOG_GENERAL(WARNING,
+                  "Exception thrown trying to reset storage " << e.what());
+    }
+    // If Instructed to reset the Code do so and call SetImmutable to reset
+    // the hash
+    try {
+      if (op.Code().size() > 0)
+        contractAccount->SetImmutable(
```

Comment:
Again, multiple Ethereum CREATE instructions can produce multiple new contracts, each with their own address and code, nothing specific to current `contractAccount`